include/albatross/src/core/linear_combination.hpp

```diff
@@ -74,6 +74,63 @@ inline auto difference(const X &x, const Y &y) {
   return difference(vx, vy);
 }
 
+template <typename Derived, typename X>
+inline auto operator*(const Eigen::SparseMatrixBase<Derived> &matrix,
+                      const std::vector<X> &features) {
+  std::vector<LinearCombination<X>> output(cast::to_size(matrix.rows()));
+
+  std::vector<std::vector<double>> output_coefs(cast::to_size(matrix.rows()));
+
+  for (Eigen::Index k = 0; k < matrix.outerSize(); ++k) {
+    for (typename Derived::InnerIterator it(matrix.derived(), k); it; ++it) {
+      output[cast::to_size(it.row())].values.push_back(
```

Comment:
nit: might be easier to read if you define separate variables for `std::vector` indices.